### PR TITLE
fix(1925): support branch names with number sign

### PR DESCRIPTION
**Pull request by wahapo**
## Context
This PR is for this Issue.
https://github.com/screwdriver-cd/screwdriver/issues/1925

Number sign of the branch name must be encoded to `%23`.

**Current**

**After**

Link to below page.

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: tests/integration/helpers/branch-url-encode-test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+
+module('helper:branch-url-encode', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders a duration given two parsable times in HH:mm:ss format', async function(assert) {
```
Comment (ibu1224) [inline tests/integration/helpers/branch-url-encode-test.js]: The description is wrong?

File: app/helpers/branch-url-encode.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * @method branchUrlEncode
+ * @param  {Array}     url      url
+ * @return {String}             encoded url
+ */
+export function branchUrlEncode([url]) {
+  if (!url) {
+    return url;
+  }
+
+  return url.replace(/#/g, '%23');
```
Comment (wahapo) [inline app/helpers/branch-url-encode.js]: Only the `#` in the branch name needs to be converted.
In the `url` shown here, the anchor `#` is not used, so we should be able to replace all `#` without any problems.

File: tests/integration/helpers/branch-url-encode-test.js (hunk 1 above)
Comment (wahapo) [reply tests/integration/helpers/branch-url-encode-test.js]: thanks, I fixed.

**Comment by yoshwata** [comment]
[ask] Does this PR works properly when pipeline uses SourceDirectory?
https://docs.screwdriver.cd/user-guide/configuration/sourceDirectory.html

And how it displayed as branch name?

**Comment by wahapo** [comment]
@yoshwata 
Also in that case, properly build was executed. And the link seems to be as expected.

Link to the GitHub page below.

**Pull request merged**